### PR TITLE
GI-70 add default status of draft for ideas

### PR DESCRIPTION
**Pull request by davidkelliott**
We want to know which ideas have not been approved (ie and admin has
approved them) we have an awaiting_approval status for ideas which have
been submitted but there was no status for initially created ideas.  By
giving new ideas a draft default status, it is much clearer.

### CODE REVIEW

File: app/models/idea.rb [hunk 1]
```diff
@@ -97,6 +97,7 @@ def approved_by_admin?
     interim_benefits
     benefits_realised
     not_proceeding
+    draft
```
Comment (heidar) [inline app/models/idea.rb]: Shall we switch to the hash form so we can have these in order?

```
enum status: {
  draft: 0,
  awaiting_approval: 1,
  ...
```

File: app/views/ideas/show.html.erb [hunk 2]
```diff
@@ -65,7 +65,7 @@
 
 <%= link_to 'Edit', edit_idea_path(@idea) %> |
 <%= link_to 'Back', ideas_path %>
-<% if @idea.status == nil %>
+<% if @idea.status == 'draft' %>
```
Comment (heidar) [inline app/views/ideas/show.html.erb]: You could do `@idea.draft?` here

File: db/migrate/20190114094201_add_default_status_to_ideas.rb [hunk 3]
```diff
@@ -0,0 +1,7 @@
+# frozen_string_literal: true
+
+class AddDefaultStatusToIdeas < ActiveRecord::Migration[5.2]
+  def change
+    change_column_default :ideas, :status, Idea.statuses[:draft]
```
Comment (heidar) [inline db/migrate/20190114094201_add_default_status_to_ideas.rb]: 💯

File: db/schema.rb [hunk 4]
```diff
@@ -40,7 +40,7 @@
     t.integer "user_id"
     t.datetime "submission_date"
     t.integer "assigned_user_id"
-    t.integer "status"
+    t.integer "status", default: 7
```
Comment (heidar) [inline db/schema.rb]: I'd refer to `Idea.statuses[:draft]` here too

File: spec/requests/ideas_spec.rb [hunk 5]
```diff
@@ -38,6 +38,7 @@
         end.to change(Idea, :count).by(1)
 
         expect(response).to have_http_status(302)
+        expect(idea.status).to eq 'draft'
```
Comment (heidar) [inline spec/requests/ideas_spec.rb]: Might be nicer as:

`expect(idea.draft?).to be_true`

File: app/models/idea.rb [hunk 6]
```diff
@@ -19,7 +19,7 @@ def submitted?
   end
 
   def approved_by_admin?
-    return false if status == 'awaiting_approval' || status.nil?
+    return false if status == 'awaiting_approval' || status == 'draft'
```
Comment (heidar) [inline app/models/idea.rb]: `status == 'awaiting_approval' || status == 'draft'`
Could be rewritten as
`awaiting_approval? || draft?`

**Comment by davidkelliott** [comment]
Thanks @heidar I've made the suggested changes 👍

**Pull request merged**